### DISCUSSION TITLE
How to use comments with i18n._()?

### DISCUSSION THREAD

**Discussion by iynaix** [lingui/js-lingui#1381] [Q&A] [answered]
How do I have a comment extracted into the .po file using `i18n._()`?

I have tried:

```typescript
i18n._({
    id: "Code",
    message: "Code",
    comment: "2 or 3 letter abbreviation",
})
```

However, I'm getting errors such as the following:

```
Missing message ID, skipping.
i18n._({
  id: "Code",
  message: "Code",
  comment: "2 or 3 letter code for an event"
})
```

I feel like I'm missing something incredibly obvious. (not using the extraction macros in this codebase as babel macros are incompatible with NextJS swc compiler)

**Comment by timofei-iatsenko** [comment, marked as answer]
You can use SWC plugin. 

https://github.com/lingui/swc-plugin

Using linguijs without a macros is discouraged.

**Reply by iynaix** [reply]
Wow, i was unaware that a swc plugin existed :smiley:, gonna use that then!

**Reply by timofei-iatsenko** [reply]
Yes, it should be promoted better. We are planing to work on that

**Comment by timofei-iatsenko** [comment]
What about your code snippet it doesn't work because it should be either: 

```ts
// supposed for "manual" usage, extractor would match i18n._ call expression
i18n._("Code", {
    message: "Code",
    comment: "2 or 3 letter abbreviation",
})
```

or

```ts
// This is generated by macro, 
// note the /* i18n */ comment. Object Expression following this comment would be
// treated and extracted as a message descriptor (no matter what fn it was passed to)
i18n._(/*i18n*/{
  id: "Code",
  message: "Code",
  comment: "2 or 3 letter code for an event"
})
```
